java/client/src/test/java/io/vitess/client/RpcClientTest.java

```diff
@@ -144,6 +150,45 @@ public void setUp() {
     return values;
   }
 
+  /**
+   * waitForVtgateclienttest waits up to 60 seconds for the "vtgateclienttest" binary to come up
+   * and be reachable on the RPC port.
+   *
+   * We use the "GetSrvKeyspace" RPC exemplary to test if vtgate responds.
+   *
+   * @throws SQLException
+   * @throws InterruptedException
+   */
+  private void waitForVtgateclienttest() throws SQLException, InterruptedException {
+    DateTime start = DateTime.now();
+    DateTime deadline = start.plusSeconds(60);
+
+    boolean waited = false;
+    while (DateTime.now().isBefore(deadline)) {
+      try {
+        ctx = Context.getDefault().withDeadlineAfter(Duration.millis(5000)).withCallerId(CALLER_ID);
```

Comment:
Harshit already gave the answer back then:

> AFAIK, The timeout is not used in the connection creation. It is only used in execute calls.

What he meant was that `create()` ignores `ctx`, but our regular RPCs respect them.

The `ctx` here has a deadline to avoid that the RPC hangs for too long.

Probably for the same reason, all RPCs in the test use a Context with a deadline as well. I've pushed an updated commit where I've added a comment in `setUp()` to explain the reasoning and the fact that you shouldn't share Context objects in general.